velox/common/hyperloglog/Murmur3Hash128.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/hyperloglog/Murmur3Hash128.h"
+
+#include "velox/common/base/Exceptions.h"
+
+namespace facebook::velox::common::hll {
+
+namespace {
+const uint64_t C1 = 0x87c37b91114253d5L;
+const uint64_t C2 = 0x4cf5ad432745937fL;
+} // namespace
+
+int64_t getLong(const void* data, int32_t offset) {
+  return *(int64_t*)((char*)data + offset);
```

Comment:
This is not safe, use `folly::loadUnaligned` instead

Reply:
Hi @Yuhta, wondering why we prefer to use folly::loadUnaligned? I saw the code of this method create a variable of the target type and do a memcpy of the input value to it. Feels that we don't need the memcpy here and that could be a performance concern?

Reply:
The memcpy should be optimized away, there is some risk the process would crash if the `data` is not aligned correctly if we just do the reinterpret cast